api/types/accesslist/member.go

```diff
@@ -22,11 +22,16 @@ import (
 	"github.com/gravitational/trace"
 
 	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/compare"
 	"github.com/gravitational/teleport/api/types/header"
 	"github.com/gravitational/teleport/api/types/header/convert/legacy"
 	"github.com/gravitational/teleport/api/utils"
 )
 
+var (
+	_ compare.IsEqual[*AccessListMember] = (*AccessListMember)(nil)
+)
```

Comment:
nit:

```suggestion
var _ compare.IsEqual[*AccessListMember] = (*AccessListMember)(nil)
```

I'm not a fan of these kinds of type assertions, I think usage should naturally dictate coercion to interfaces. That said, feel free to resolve as you prefer.